apps/wing-console/console/ui/src/shared/use-vscode-links.tsx

```diff
@@ -0,0 +1,51 @@
+import { PropsWithChildren } from "react";
+
+import { trpc } from "../services/trpc.js";
+
+export const useVSCodeLinks = () => {
+  const createHtmlLink = (
+    error: string,
+    className: string,
+    expanded: boolean = false,
+  ) => {
+    return error
+      .replaceAll(
+        /\B((?:[a-z]:)?[/\\]\S+):(\d+):(\d+)/gi,
+        (match, path, line, column) => {
+          const link = `vscode://file/${path}:${line}:${column}`;
+          return `<a class="${className}" href="${link}">${match}</a>`;
```

Comment:
We can use data attributes on the element to store the path, line and columns separately. Then, in the `OpenFileInEditorButton` you only need to retrieve them instead of parsing them from the link.